functions/executor/call.py

```diff
@@ -113,22 +115,26 @@ def _exec_dag_function_normal(pusher_cache, kvs, triggers, function, schedule):
 
 def _exec_func_normal(kvs, func, args):
     func_args = ()
-
     to_resolve = []
+    deserialize = {}
 
     # resolve any references to KVS objects
     key_index_map = {}
     for i, arg in enumerate(args):
         if isinstance(arg, FluentReference):
             to_resolve.append(arg)
             key_index_map[arg.key] = i
+            deserialize[arg.key] = arg.deserialize
         func_args += (arg,)
 
     if len(to_resolve) > 0: 
         kv_pairs = _resolve_ref_normal(to_resolve, kvs)
 
         for key in kv_pairs:
-            func_args[key_index_map[key]] = kv_pairs[key]
+            if deserialize[key]:
+                func_args[key_index_map[key]] = deserialize_val(kv_pairs[key])
```

Comment:
Below, you pass in `kv_pairs[key][1]`, but here, it's `kv_pairs[key]`? Are the pairs key-value pairs? What is the type of the value? I assume a `LWWPairLattice`. In which case, we'd need to pass in `kv_pairs[key].reveal()[1]`.